aws/resource_aws_glue_user_defined_function.go

```diff
@@ -0,0 +1,259 @@
+		DatabaseName: aws.String(dbName),
+		FunctionName: aws.String(funcName),
+	}
+
+	out, err := conn.GetUserDefinedFunction(input)
+	if err != nil {
+
+		if isAWSErr(err, glue.ErrCodeEntityNotFoundException, "") {
+			log.Printf("[WARN] Glue User Defined Function (%s) not found, removing from state", d.Id())
+			d.SetId("")
+			return nil
+		}
+
+		return fmt.Errorf("error reading Glue User Defined Function: %w", err)
+	}
+
+	udf := out.UserDefinedFunction
+
+	udfArn := arn.ARN{
+		Partition: meta.(*AWSClient).partition,
+		Service:   "glue",
+		Region:    meta.(*AWSClient).region,
+		AccountID: meta.(*AWSClient).accountid,
+		Resource:  fmt.Sprintf("userDefinedFunction/%s/%s", dbName, aws.StringValue(udf.FunctionName)),
+	}.String()
+
+	d.Set("arn", udfArn)
+	d.Set("name", udf.FunctionName)
+	d.Set("catalog_id", catalogID)
+	d.Set("database_name", dbName)
+	d.Set("owner_type", udf.OwnerType)
+	d.Set("owner_name", udf.OwnerName)
+	d.Set("class_name", udf.ClassName)
+	if udf.CreateTime != nil {
+		d.Set("create_time", udf.CreateTime.Format(time.RFC3339))
+	}
+	if err := d.Set("resource_uris", flattenAwsGlueUserDefinedFunctionResourceUri(udf.ResourceUris)); err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func resourceAwsGlueUserDefinedFunctionDelete(d *schema.ResourceData, meta interface{}) error {
+	conn := meta.(*AWSClient).glueconn
+	catalogID, dbName, funcName, err := readAwsGlueUDFID(d.Id())
+	if err != nil {
+		return err
+	}
+
+	log.Printf("[DEBUG] Glue User Defined Function: %s", d.Id())
+	_, err = conn.DeleteUserDefinedFunction(&glue.DeleteUserDefinedFunctionInput{
+		CatalogId:    aws.String(catalogID),
+		DatabaseName: aws.String(dbName),
+		FunctionName: aws.String(funcName),
+	})
+	if err != nil {
+		return fmt.Errorf("error deleting Glue User Defined Function: %w", err)
+	}
+	return nil
+}
+
+func readAwsGlueUDFID(id string) (catalogID string, dbName string, funcName string, err error) {
+	idParts := strings.Split(id, ":")
+	if len(idParts) != 3 {
+		return "", "", "", fmt.Errorf("unexpected format of ID (%q), expected CATALOG-ID:DATABASE-NAME:FUNCTION-NAME", id)
+	}
+	return idParts[0], idParts[1], idParts[2], nil
+}
+
+func expandAwsGlueUserDefinedFunctionInput(d *schema.ResourceData) *glue.UserDefinedFunctionInput {
+
+	udf := &glue.UserDefinedFunctionInput{
+		ClassName:    aws.String(d.Get("class_name").(string)),
+		FunctionName: aws.String(d.Get("name").(string)),
+		OwnerName:    aws.String(d.Get("owner_name").(string)),
+		OwnerType:    aws.String(d.Get("owner_type").(string)),
+	}
+
+	if v, ok := d.GetOk("resource_uris"); ok && v.(*schema.Set).Len() > 0 {
+		udf.ResourceUris = expandAwsGlueUserDefinedFunctionResourceUri(d.Get("resource_uris").(*schema.Set))
+	}
+
+	return udf
+}
+
+func expandAwsGlueUserDefinedFunctionResourceUri(conf *schema.Set) []*glue.ResourceUri {
+	result := make([]*glue.ResourceUri, 0, conf.Len())
+
+	for _, r := range conf.List() {
+		uriRaw := r.(map[string]interface{})
```

Comment:
To prevent potential panics, should verify the type assertion here:

```suggestion
		uriRaw, ok := r.(map[string]interface{})

		if !ok {
			continue
		}
```